docs/proposals/diamond-design.md

```diff
@@ -0,0 +1,458 @@
+The bundle on which conflicts where initially detected remains untouched for traceability.
+
+### Cancel a diamond
+```
+datamon bundle diamond cancel --repo {repo} --diamond {diamond-id}
+```
+
+This waits for ongoing running splits to complete then marks vmetadata ready for cleanup
+
+**NOTE**: in the future, we may interrupt running splits rather than wait for them to complete _then_ cancel the dimaond
+
+### Identify splits with some custom node identifier
+```
+datamon bundle diamond split add --repo {repo} --path {source} --split-tag {my-bespoke-distinctive-tag}
+```
+
+Tags are kept in metadata and can be used when reporting about running splits or help latter with conflict resolution.
+Tags are typically a hostname or something that helps the end-user to identify a node in the diamond.
+
+## Nice to have / future
+
+   1. `add` commands can be strongly ordered based on a design similar to WAL (KSUIDs alone do not guarantee strong ordering).
+      This is a nice to have addition and can be added later in the implementation.
+   2. explicit cancellation could attempt to interrupt ongoing splits
+
+## Recap of new commands or new options to existing commands
+
+```
+datamon bundle diamond
+              |_  initialize --repo {repo}
+              |
+              |_  commit --repo {repo} --diamond {diamond-id} --message {commit message} [--no-conflict] [--label]
+              |
+              |_  list --repo {repo}
+              |
+              |_  cancel --repo {repo} --diamond {diamond-id}
+              |
+              |_  split
+                    |
+                    |_ add --repo {repo} --diamond {diamond-id} --path {source} [--split-tag {my-bespoke-distinctive-tag}] [--files {list of files}]
+                    |
+                    |_ list --repo {repo} --diamond {diamond-id}
+
+
+datamon bundle conflicts resolve --bundle {bundle-id} {--files {file-list}|--resolve-all} [--message {...} --label {...}]
+
+datamon bundle get --repo {repo} --bundle {bundle-id} [--with-conflicts]
+
+datamon bundle get --repo {repo} --bundle {bundle-id} [--with-diamonds]
+
+datamon bundle list files --repo {repo} --bundle {bundle-id} [--with-conflicts|--with-checkpoints]
+
+datamon bundle download --repo {repo} --bundle {bundle-id} [--with-conflicts|--with-checkoints] --destination {local path}
+```
+
+
+## Recap of the extended metadata layout
+
+### metadata
+```
+bundles/{repo}/{bundle-id}/bundle.yaml            # <- additional audit trail information
+```
+
+### vmetadata
+
+```
+splits/{repo}/{diamond-id}/diamond.yaml            # <- captures the diamond state: initialized, committing, done
+splits/{repo}/{diamond-id}/{split-id}/split.yaml   # <- captures the split state, plus holds information about the running split, eventually merged into metadata for bundle
+splits/{repo}/{diamond-id}/{split-id}/bundle-files-{index}.yaml
+
+conflicts/{repo}/{bundle-id}/conflict-files-{index}.yaml
```

Comment:
Drop the conflicts and checkpoint, I want to keep the implementation for incremental and split as close as possible.. just allowing a ```split-id```` and not creating conflicts within a split-id will solve incremental uploads.

Reply:
I understand the militancy for —split-id but the undue creation of conflicts with my proposal is a false premise

Reply:
Implementation is thr same.conflicts and checkpoints only differ in name

Reply:
In case a pod that does not use split-id restarts and retries upload, at commit time there will be no way to know that the second file list is not in conflict. Also, the developer might want commit with no conflict in which case we will introduce a conflict where there is none.

Reply:
ok dropped the specific metadata. conflicting files are moved to a special "hidden" path at commit time.